File: Part 2 - Regression/Section 4 - Simple Linear Regression/linreg.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
Created on Mon Jul  2 11:29:11 2018

@author: ross
"""


import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

dataset = pd.read_csv('Salary_Data.csv')
x = dataset.iloc[:, :-1].values
y = dataset.iloc[:, 1].values


from sklearn.model_selection import train_test_split
x_train, x_test, y_train, y_test = train_test_split(x, y, test_size=1/3, random_state=0)

""" Linear regression does not need scaling 
from sklearn.preprocessing import StandardScaler 
sc_X = StandardScaler()
x_train = sc_X.fit_transform(x_train)
x_test = sc_X.transform(x_test)
"""

# Fit to Linear regression model
from sklearn.linear_model import LinearRegression
regressor = LinearRegression()
regressor.fit(x_train, y_train)

# Predict
y_pred = regressor.predict(x_test)

# Visualization
plt.scatter(x_train, y_train, color='red')
plt.plot(x_train, regressor.predict(x_train), color='blue')
plt.title('Salary vs Experience (Training set)')
plt.xlabel('Years')
plt.ylabel('Salary')
plt.show()

# Visualization test
plt.scatter(x_test, y_test, color='red')
plt.plot(x_train, regressor.predict(x_train), color='blue')
plt.title('Salary vs Experience (Testing set)')
plt.xlabel('Years')
plt.ylabel('Salary')
plt.show()

File: Part 2 - Regression/Section 4 - Simple Linear Regression/linreg.R
setwd('/Users/ross/Documents/udemy/Machine Learning A-Z Template Folder/Part 2 - Regression/Section 4 - Simple Linear Regression/')

dataset <- read.csv("Salary_Data.csv")
# x <- dataset[, 1:3]
# y <- dataset[, 4]

library(caTools)
set.seed(123)
split = sample.split(dataset$Salary, SplitRatio = 2/3)
training_set <- subset(dataset, split == TRUE)
test_set <- subset(dataset, split == FALSE)

# Fit simple linear regression
regressor <- lm(formula = Salary ~ YearsExperience,
                data=training_set)

# Get the predicted values
y_pred <- predict(regressor, newdata=test_set)

# plot the data
library(ggplot2)
ggplot() +
  geom_point(aes(x=training_set$YearsExperience, y=training_set$Salary),
             color='red') + 
  geom_line(aes(x=training_set$YearsExperience, y=predict(regressor, newdata=training_set)),
            color='blue') + 
  ggtitle('Salary vs experience - Training set') + 
  xlab('Years of experience') + 
  ylab('Salary')

ggplot() +
  geom_point(aes(x=test_set$YearsExperience, y=test_set$Salary),
             color='red') + 
  geom_line(aes(x=training_set$YearsExperience, y=predict(regressor, newdata=training_set)),
            color='blue') + 
  ggtitle('Salary vs experience - Test set') + 
  xlab('Years of experience') + 
  ylab('Salary')
